libs/stats/odc/stats/cli.py

```diff
@@ -82,26 +87,86 @@ def save_tasks(grid, year, period,
     from .metadata import gs_bounds, compute_grid_info, gjson_from_tasks
     from .model import DateTimeRange
 
-    if period is not None and year is not None:
-        print("Can only supply one of --year or --period", file=sys.stderr)
+    if temporal_range is not None and year is not None:
+        print("Can only supply one of --year or --temporal_range", file=sys.stderr)
         sys.exit(1)
 
-    if period is not None:
+    if temporal_range is not None:
         try:
-            period = DateTimeRange(period)
+            temporal_range = DateTimeRange(temporal_range)
         except ValueError:
-            print(f"Failed to parse supplied period: '{period}'")
+            print(f"Failed to parse supplied temporal_range: '{temporal_range}'")
             sys.exit(1)
 
     if year is not None:
-        period = DateTimeRange.year(year)
+        temporal_range = DateTimeRange.year(year)
+
+    if period is not None:
+        try:
+            period = DateTimeRange(period) if not period in ['all' , 'annual'] else period
+        except ValueError:
+            print(f"Failed to parse supplied period: '{period}'")
+            sys.exit(1)
 
     if output == '':
-        if period is not None:
-            output = f'{product}_{period.short}.db'
+        if temporal_range is not None:
+            output = f'{product}_{temporal_range.short}.db'
         else:
             output = f'{product}_all.db'
 
+    def bin_data(dss: Dict[Tuple, list]) -> Dict[Tuple, List]:
```

Comment:
That's not a valid type annotation

```python
Dict[Tuple[int,int], List[CompressedDataset]]
```

one has to specify what's inside the Tuple/List

Reply:
same goes for output type annotation

Reply:
Specified type of data elements in the annotation.